File: String/Reverse words in a given string/reverseWords.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  string reverseWords(string S) {
    int i = S.length() - 1;
    string ans;
    string temp = "";
    while (i >= 0) {
      if (S[i] == '.') {
        reverse(temp.begin(), temp.end());
        ans += (temp);
        ans += (".");
        temp = "";
      } else {
        temp += S[i];
      }
      i--;
    }
    reverse(temp.begin(), temp.end());
    ans += temp;
    return ans;
  }
};

int main() {
  int t;
  cin >> t;
  while (t--) {
    string s;
    cin >> s;
    Solution obj;
    cout << obj.reverseWords(s) << endl;
  }
}

File: String/Reverse words in a given string/ReverseWords.java
import java.util.Scanner;

class ReverseWords {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    while (t > 0) {
      String s = sc.next();
      Solution obj = new Solution();
      System.out.println(obj.reverseWords(s));
      t--;
    }
    sc.close();
  }
}

class Solution {
  String reverseWords(String S) {
    int i = S.length() - 1;
    StringBuilder ans = new StringBuilder();
    StringBuilder temp = new StringBuilder();
    while (i >= 0) {
      if (S.charAt(i) == '.') {
        ans.append(temp.reverse().toString());
        ans.append('.');
        temp.setLength(0);
      } else {
        temp.append(S.charAt(i));
      }
      i--;
    }
    ans.append(temp.reverse().toString());
    return ans.toString();
  }
}
